### Eclipse Workspace Patch 1.0
#P RoundCube
Index: index.php
===================================================================
--- index.php	(revision 2850)
+++ index.php	(working copy)
@@ -131,6 +131,7 @@
 else if ($RCMAIL->task=='logout' && isset($_SESSION['user_id'])) {
   $OUTPUT->show_message('loggedout');
   $RCMAIL->logout_actions();
+  $RCMAIL->plugins->exec_hook('logout_after');
   $RCMAIL->kill_session();
 }
 
