### Eclipse Workspace Patch 1.0
#P RoundCube
Index: program/include/rcube_imap.php
===================================================================
--- program/include/rcube_imap.php	(revision 2850)
+++ program/include/rcube_imap.php	(working copy)
@@ -112,13 +112,21 @@
     $ICL_PORT = $port;
     $IMAP_USE_INTERNAL_DATE = false;
     
-    $data = rcmail::get_instance()->plugins->exec_hook('imap_connect', array('host' => $host, 'user' => $user));
-    if (!empty($data['pass']))
-      $pass = $data['pass'];
+    $attempt = 0;
+    do {
+      $attempt++;
+      
+      $data = rcmail::get_instance()->plugins->exec_hook('imap_connect', array('attempt' => $attempt, 'host' => $host, 'user' => $user));
+      $host = $data['host'];
+      $user = $data['user'];
+      if (!empty($data['pass']))
+        $pass = $data['pass'];
 
-    $this->conn = iil_Connect($data['host'], $data['user'], $pass, $this->options);
-    $this->host = $data['host'];
-    $this->user = $data['user'];
+      $this->conn = iil_Connect($host, $user, $pass, $this->options);
+    } while (!$this->conn && $data['retry']);
+    
+    $this->host = $host;
+    $this->user = $user;
     $this->pass = $pass;
     $this->port = $port;
     $this->ssl = $use_ssl;
